src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -367,19 +367,21 @@ public static string Combine(params string[] paths)
                 if (IsPathRooted(paths[i]))
                 {
                     firstComponent = i;
-                    finalSize = paths[i].Length;
+                    maxSize = paths[i].Length;
                 }
                 else
                 {
-                    finalSize += paths[i].Length;
+                    maxSize += paths[i].Length;
                 }
 
                 char ch = paths[i][paths[i].Length - 1];
                 if (!PathInternal.IsDirectorySeparator(ch))
-                    finalSize++;
+                    maxSize++;
             }
-
-            StringBuilder finalPath = StringBuilderCache.Acquire(finalSize);
+            
+            Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
```

Comment:
PathInternal.MaxShortPath is not defined on unix. just directly use the value and add a comment that it the maxshort path on windows